lib/commands/sdk.js

```diff
@@ -549,427 +542,505 @@ SdkSubcommands.install = {
+}
+
+/**
+ * @param {object} logger
+ * @param {object} config
+ * @param {object} cli
+ * @param {string} version version to install
+ * @param {string} osName
+ * @throws {Error} if no releases are found
+ * @returns {Promise<SDKInstallRequest>}
+ */
+async function doReleases(logger, config, cli, version, osName) {
+	// check if it's a valid release
+	const releases = await getReleases(config, osName);
+	if (!releases || releases.size === 0) {
+		throw new Error(__('No releases found!') + '\n');
+	}
+
+	const isLatest = !version || version === 'latest';
+	const rels = Array.from(releases.keys()).sort().reverse();
+
+	// if choosing latest, resolve to latest listed releases
+	if (isLatest) {
+		version = rels[0];
+	}
+
+	if (releases.has(version)) {
+		// we found the version, move on...
+		const release = releases.get(version);
+		return new SDKInstallRequest(logger, config, cli, {
+			url: release.url, // TODO: Use rest of metadata we have like checksum for integrity check?
+			version,
+			isLatest,
+			setDefault: isLatest // we want to auto-select it if it's the latest GA
 		});
 	}
-};
+
+	logger.log(__('Did not find a release %s, scanning branches...', version.cyan));
+	const matchingBuild = await scanBranches(logger, config, cli, version, osName);
+	if (!matchingBuild) {
+		let str = __('Release "%s" does not exist', version) + '\n';
+		if (!cli.argv.branch && version.indexOf('.v') !== -1) {
+			str += __('Did you forget to specify the branch?') + '\n';
+		}
+		appc.string.suggest(version, rels, s => {
+			str += (s || '') + '\n';
+		}, 1);
+		str += __('Available Releases:');
+		str += appc.string.renderColumns(rels, '    ', 100).cyan + '\n';
+		throw new Error(str);
+	}
+
+	logger.log(__('Found build %s in branch %s', version.cyan, matchingBuild.branch.cyan) + '\n');
+	return matchingBuild;
+}
+
+/**
+ * @param {string} branch branch name to validate/get, use 'latest' to pick default branch
+ * @param {object} config cli config
+ * @returns {Promise<string>} validated/resolved branch name
+ * @throws {Error} if no branches are found, or if named barnch is not found in listing
+ */
+async function getBranch(branch, config) {
+	const data = await fetch(urls.branches, config);
+
+	// check that we have branches
+	if (!data || !data.branches.length) {
+		throw new Error(__('No branches found!') + '\n');
+	}
+
+	// resolve 'latest' alias
+	if (branch === 'latest') {
+		branch = data.defaultBranch;
+	}
+
+	// check that the desired branch exists
+	if (!data.branches.includes(branch)) {
+		let str = __('Branch "%s" does not exist', branch) + '\n';
+		appc.string.suggest(branch, data.branches, s => {
+			str += (s || '') + '\n';
+		}, 2);
+		str += __('Available Branches:');
+		str += appc.string.renderColumns(data.branches.sort().reverse(), '    ', 100).cyan + '\n';
+		throw new Error(str);
+	}
+	return branch;
+}
+
+/**
+ * @param {object} logger
+ * @param {object} config
+ * @param {object} cli
```

Comment:
<!--
  0 failure: 
  1 warning:  lib/commands/sdk....
  
  
  DangerID: danger-id-Danger;
  File: lib/commands/sdk.js;
  Line: 982;
-->

- :warning: lib/commands/sdk.js line 982 – Missing JSDoc parameter description for 'cli'. (valid-jsdoc)